Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -1721,6 +1723,27 @@ private extension Purchases {
         }
 
         self.updateOfferingsCache(isAppBackgrounded: isAppBackgrounded)
+
+        self.fetchAndCacheAppTransactionIfNeeded()
+    }
+
+    func fetchAndCacheAppTransactionIfNeeded() {
+        if self.systemInfo.dangerousSettings.internalSettings.usesStoreKit2JWS,
+           self.systemInfo.originalAppVersion == nil,
+           self.systemInfo.originalAppPurchaseDate == nil {
+            if #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *) {
+                _ = Task<Void, Never> {
+                    if let appTransaction = await self.transactionFetcher.appTransaction {
+                        systemInfo.originalAppVersion = appTransaction.originalApplicationVersion
+                        systemInfo.originalAppPurchaseDate = appTransaction.originalPurchaseDate
+                    } else {
+                        let receipt = try? await self.fetchReceipt(.onlyIfEmpty)
```

Comment:
⚠️ (cc @aboedo so he can confirm), but if we use `.onlyIfEmpty`, users would see a login prompt if the receipt is empty. We need to use `.never`